autocodesign/mock_AssetWriter.go

```diff
@@ -0,0 +1,24 @@
+// Code generated by mockery 2.9.4. DO NOT EDIT.
```

Comment:
Our convention for adding autogenerated mocks is to create a `mocks` subpackge where a thing is mocked, for example `autocodesign/mocks/AssetWriter` in this case.

This way our check step will exclude the mock files from errchecking: https://github.com/bitrise-steplib/steps-check/pull/6

Reply:
Can not do that as it leads to an import cycle (as there are types defined in the package that are used in the interfaces)